datatree/io.py

```diff
@@ -146,6 +139,11 @@ def _datatree_to_netcdf(
     if encoding is None:
         encoding = {}
 
+    if set(encoding) - set(dt.groups):
+        raise ValueError(
+            f"unexpected encoding group name(s) provided: {set(encoding) - set(dt.groups)}"
+        )
```

Comment:
I think something like this is needed to avoid silently skipping paths that are intended to be included.

Reply:
We probably want to generalize this check into a utility function that can be called on any dict of tree paths, and maybe also check for the opposite case (groups for which no encoding was specified).

Reply:
I don't think we want to check for the opposite case (groups for which no encoding was specified). We also need this check in `to_zarr` which I have now added. I didn't split this out into a separate function since its so simple. Open to suggestions though.